lib/deferredPromise.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var Promise = require('bluebird');
+
+/** A helper function for creating a deferred promise.
+ * This should only be used when using an API that does not support
+ * promises, such as loading an HTMLImageElement, or using Google's NodeJS API.
+ *
+ * @return {Object} An object with a promise, and reject and resolve functions.
+ */
+module.exports = function () {
```

Comment:
deferreds are considered an anti-pattern, I would delete this file and just use `new Promise`.